src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -244,6 +274,30 @@ class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
       clearItems(this.hass).catch(() => this._fetchData());
     }
   }
+
+  private _addItem(ev): void {
+    if (this.shadowRoot!.querySelector(".addBox")) {
```

Comment:
No need for if-check, it can't be triggered unless it's rendered?

Also prefer if we put the selectors in individual getters. Helps with TypeScript mangling and abstracts that away: https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/editor/hui-dialog-edit-card.ts#L65-L67